server-extension/src/main/java/flyway/paikkis/V2_18__link_selkokartta_to_bglayer_selector.java

```diff
@@ -0,0 +1,108 @@
+package flyway.paikkis;
+
+import fi.nls.oskari.log.LogFactory;
+import fi.nls.oskari.log.Logger;
+import org.flywaydb.core.api.migration.jdbc.JdbcMigration;
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+import fi.nls.oskari.util.JSONHelper;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+
+// Note to English speakers: "Selkokartta" is a background layer for cognitively impaired users
+
+public class V2_18__link_selkokartta_to_bglayer_selector implements JdbcMigration {
+    private final Logger LOG = LogFactory.getLogger(this.getClass());
+
+    private static final String EXTERNAL_ID = "base_selko";
+    private static final String PLUGIN_ID = "Oskari.mapframework.bundle.mapmodule.plugin.BackgroundLayerSelectionPlugin";
+
+    public void migrate(Connection connection) throws Exception {
+        Integer selkokarttaId = getSelkoLayer(connection);
+        if (selkokarttaId == null) {
+            LOG.warn("No Selkokartta layer found, skipping rest of migration!");
+            return;
+        }
+        updateLayerExternalId(connection, selkokarttaId, EXTERNAL_ID);
+
+        updateViewsWithBaselayerSelector(connection, EXTERNAL_ID);
+    }
+
+    private Integer getSelkoLayer(Connection conn) throws SQLException {
+        // find first Selkokartta layer
+        final String sql = "SELECT id FROM oskari_maplayer WHERE name = 'selkokartta' AND url LIKE 'https://karttamoottori.maanmittauslaitos.fi%' LIMIT 1";
+        try (PreparedStatement statement = conn.prepareStatement(sql)) {
+            try (ResultSet rs = statement.executeQuery()) {
+                if (rs.next()) {
+                    return rs.getInt("id");
+                }
+            }
+        }
+        return null;
+    }
+
+    private void updateLayerExternalId(Connection conn, int layerId, String externalId) throws SQLException {
+        final String sql = "UPDATE oskari_maplayer SET externalid=? WHERE id=?";
+        try (PreparedStatement statement = conn.prepareStatement(sql)) {
+            statement.setString(1, externalId);
+            statement.setInt(2, layerId);
+            statement.execute();
+        }
+    }
+
+    private void updateViewsWithBaselayerSelector(Connection conn, String externalId) throws SQLException {
+        final String sql = "SELECT view_id, seqno, config FROM portti_view_bundle_seq WHERE bundleinstance = 'mapfull'";
```

Comment:
Change this to:

     SELECT view_id, seqno, config FROM portti_view_bundle_seq WHERE bundle_id = (select id from portti_bundle name='mapfull')

Just to be sure